src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Config/RemoveAzureVMDataDiskCommand.cs

```diff
@@ -39,9 +39,8 @@ public class RemoveAzureVMDataDiskCommand : Microsoft.Azure.Commands.ResourceMan
 
         [Alias("Name")]
         [Parameter(
-            Mandatory = true,
+            Mandatory = false,
             Position = 1,
-            ValueFromPipelineByPropertyName = true,
```

Comment:
Actually there is an issue with pipeline with Remove-AzureRmVMDataDisk (and other Remove command that removes a property from VM object.)  Since this parameteter (DataDiskNames) have an alias "Name" and VM object has "Name" property for VM Name, in piping scenario (e.g. Get-AzureRmVM | Remove-AzureRmVMDataDisk ), VM name is piped to Remove-AzureRmVMDataDisk as a data disk name, and it tries to remove a data disk named VM name.   Setting ValueFromPipelineByPropertyName to false removes this issue.  Also we remove "DataDiskNames" (which is redundant property) from VM object, DataDiskNames are not piped from VM object to Remove-AzureRmVM.    Remove-AzureRmVM is also modified when DataDiskNames are not provided, it interprets it as removing all existing data disks.  So the command does not change the behavior of Get-AzureRmVM | Remove-AzureRmVM .